src/primitives/transaction.h

```diff
@@ -562,34 +591,47 @@ class CTransaction
             fOverwintered &&
             nVersionGroupId == OVERWINTER_VERSION_GROUP_ID &&
             nVersion == OVERWINTER_TX_VERSION;
+
         bool isSaplingV4 =
             fOverwintered &&
             nVersionGroupId == SAPLING_VERSION_GROUP_ID &&
             nVersion == SAPLING_TX_VERSION;
-        if (fOverwintered && !(isOverwinterV3 || isSaplingV4)) {
+
+        // It is not possible to make the transaction's serialized form vary on
+        // a per-enabled-feature basis.The approach here is that all
+        // serialization rules for not-yet-released features must be
+        // non-conflicting and transaction version/group must be set to
+        // ZFUTURE_TX_(VERSION/GROUP_ID)
```

Comment:
These `isFuture` checks will be removed as part of the changes that come with ZIP 225. Going forward, I think that it probably makes more sense to write a separate parser for each transaction version, rather than having a bunch of branching logic within a single parser.

Reply:
For ZIP 225, there will be separate parsers.